### PR TITLE
[improve](array-funcs) accelerate arrays_overlap by inverted index (#41161)

### PR DESCRIPTION
**Pull request by amorynan**
support arrays_overlap with invertedIndex 

## Proposed changes

Issue Number: close #xxx

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 1]
```diff
@@ -128,6 +129,87 @@ class FunctionArraysOverlap : public IFunction {
...
+            segment_v2::InvertedIndexReaderType::FULLTEXT) {
+            return Status::Error<ErrorCode::INVERTED_INDEX_EVALUATE_SKIPPED>(
+                    "Inverted index evaluate skipped, FULLTEXT reader can not support "
+                    "array_overlap");
+        }
+        // in arrays_overlap param is array Field and const Field
+        ColumnPtr arg_column = arguments[0].column;
+        DataTypePtr arg_type = arguments[0].type;
+        if ((is_column_nullable(*arg_column) && !is_column_const(*remove_nullable(arg_column))) ||
+            !is_column_const(*arg_column)) {
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_arrays_overlap.h:159]: logic error: if arg_column is nullable , it's always true.

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 2]
```diff
@@ -128,6 +129,87 @@ class FunctionArraysOverlap : public IFunction {
         return make_nullable(std::make_shared<DataTypeUInt8>());
     }
 
+    /**
+     * eval inverted index. we can filter array rows with inverted index iter
+     * array_overlap(array, []) -> array_overlap(array, const value)
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_arrays_overlap.h:134]: What do you mean by `array_overlap(array, []) -> array_overlap(array, const value)` ?

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 3]
```diff
@@ -128,6 +129,87 @@ class FunctionArraysOverlap : public IFunction {
...
+                roaring->addRange(0, num_rows);
+                break;
+            } else if (st != Status::OK()) {
+                return st;
+            }
+            *roaring |= *single_res;
+        }
+
+        segment_v2::InvertedIndexResultBitmap result(roaring, null_bitmap);
+        bitmap_result = result;
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_arrays_overlap.h:207]: Should you overrite bitmap_result or & ?

File: be/src/vec/functions/array/function_arrays_overlap.h [hunk 4]
```diff
@@ -128,6 +129,87 @@ class FunctionArraysOverlap : public IFunction {
...
+        for (size_t i = 0; i < query_val.size(); ++i) {
+            Field nested_query_val = query_val[i];
+            std::shared_ptr<roaring::Roaring> single_res = std::make_shared<roaring::Roaring>();
+            RETURN_IF_ERROR(InvertedIndexQueryParamFactory::create_query_value(
+                    nested_param_type, &nested_query_val, query_param));
+            Status st = iter->read_from_inverted_index(
+                    data_type_with_name.first, query_param->get_value(),
+                    segment_v2::InvertedIndexQueryType::EQUAL_QUERY, num_rows, single_res);
+            if (st.code() == ErrorCode::INVERTED_INDEX_NO_TERMS) {
+                // if analyzed param with no term, we do not filter any rows
```
Comment (xiaokang) [inline be/src/vec/functions/array/function_arrays_overlap.h:196]: Why do you need `analyze` for array_overlaps ?

File: be/src/vec/functions/array/function_arrays_overlap.h (hunk 2 above)
Comment (amorynan) [reply be/src/vec/functions/array/function_arrays_overlap.h:134]: we now just support arrays_overlap with second param is const value

File: be/src/vec/functions/array/function_arrays_overlap.h (hunk 4 above)
Comment (amorynan) [reply be/src/vec/functions/array/function_arrays_overlap.h]: I remove this , see also as array_contains

File: be/src/vec/functions/array/function_arrays_overlap.h (hunk 1 above)
Comment (amorynan) [reply be/src/vec/functions/array/function_arrays_overlap.h]: done !

**Comment by amorynan** [comment]
run buildall

**Pull request merged**